src/pudl/transform/ferc1.py

```diff
@@ -1473,1553 +1533,101 @@ def add_calculation_correction_components(
-                ],
-            },
-            "plant_in_service_ferc1": {
-                "electric_plant_in_service_and_completed_construction_not_classified_electric": [
-                    {
-                        "calc_component_to_replace": {},
-                        "calc_component_new": {
-                            "name": "general_plant",
-                            "weight": 1.0,
-                            "source_tables": ["plant_in_service_ferc1"],
-                        },
-                    },
-                    {
-                        "calc_component_to_replace": {},
-                        "calc_component_new": {
-                            "name": "transmission_and_market_operation_plant_regional_transmission_and_market_operation_plant",
-                            "weight": 1.0,
-                            "source_tables": ["plant_in_service_ferc1"],
-                        },
-                    },
-                    {
-                        "calc_component_to_replace": {},
-                        "calc_component_new": {
-                            "name": "distribution_plant",
-                            "weight": 1.0,
-                            "source_tables": ["plant_in_service_ferc1"],
-                        },
-                    },
-                    {
-                        "calc_component_to_replace": {},
-                        "calc_component_new": {
-                            "name": "transmission_plant",
-                            "weight": 1.0,
-                            "source_tables": ["plant_in_service_ferc1"],
-                        },
-                    },
-                    {
-                        "calc_component_to_replace": {},
-                        "calc_component_new": {
-                            "name": "production_plant",
-                            "weight": 1.0,
-                            "source_tables": ["plant_in_service_ferc1"],
-                        },
-                    },
-                    {
-                        "calc_component_to_replace": {},
-                        "calc_component_new": {
-                            "name": "intangible_plant",
-                            "weight": 1.0,
-                            "source_tables": ["plant_in_service_ferc1"],
-                        },
-                    },
-                ],
-            },
-        }
-
-        def remove_nones_in_list(list_of_things):
-            """Remove None's (or False's) in a list.
-
-            We're about to introduce None's into the calculation components when an
-            existing component needed to be removed.
-            """
-            return [i for i in list_of_things if i]
+        calc_comp_idx = [
+            "table_name_parent",
+            "xbrl_factoid_parent",
+            "table_name",
+            "xbrl_factoid",
+        ]
+        calc_fixes = calc_fixes.set_index(calc_comp_idx).sort_index()
```

Comment:
add assertions about index uniqueness for the future possibility of many dimensions per calc_comp_idx